### PR TITLE
Fixed #24607 -- Serialized natural keys in multi-table inheritance models.

### PR DESCRIPTION
**Pull request by timgraham**
Fixed from https://github.com/django/django/pull/8370, test from 74a575eb7296fb04e1fc2bd4e3f68dee3c66ee0a.

### CODE REVIEW

File: django/core/serializers/base.py [hunk 1]
```diff
@@ -82,8 +82,13 @@ def serialize(self, queryset, *, stream=None, fields=None, use_natural_foreign_k
             # Use the concrete parent class' _meta instead of the object's _meta
             # This is to avoid local_fields problems for proxy models. Refs #17717.
             concrete_model = obj._meta.concrete_model
+            # When using natural primary keys and multi-table inheritance,
+            # retrieve the pk field of the parent. That field must be
+            # serialized, otherwise deserialization isn't possible.
+            pk = self.use_natural_primary_keys and concrete_model._meta.pk
+            pk_parent = pk and pk.remote_field and pk.remote_field.parent_link and pk
```
Comment (felixxm) [inline django/core/serializers/base.py]: `pk` is twice. I think the last `and pk` can be removed.

File: tests/serializers/models/multi_table.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        return self.get(parent_data=parent_data)
+
+
+class Parent(models.Model):
+    parent_data = models.CharField(max_length=30, unique=True)
+
+    objects = ParentManager()
+
+    def natural_key(self):
+        return (self.parent_data, )
```
Comment (felixxm) [inline tests/serializers/models/multi_table.py]: The trailing space can be removed.

File: django/core/serializers/base.py (hunk 1 above)
Comment (timgraham) [reply django/core/serializers/base.py]: It's correct, but probably the code is too confusing. An alternate formulation is:
```
if self.use_natural_primary_keys:
    pk = concrete_model._meta.pk
    pk_parent = pk if pk.remote_field and pk.remote_field.parent_link else None
else:
    pk_parent = None
```

Comment (felixxm) [reply django/core/serializers/base.py]: Thanks. You're right too much `and` 😄. The second version is more readable for me.

**Pull request merged**